### PR TITLE
Support AWS v4 signatures

### LINKED ISSUE #24: AWS signature V4
Amazon allows V2 signatures for old regions and requires [V4 signatures](http://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html) for new regions like eu-central-1 (Frankfurt).  jclouds plans to move to V4 by default for aws-s3 [JCLOUDS-480](https://issues.apache.org/jira/browse/JCLOUDS-480) so we must support V4 to allow jclouds applications to use S3Proxy.

### PR DESCRIPTION
**Pull request by andrewgaul**
Fixes #24.

### CODE REVIEW

**Comment by andrewgaul** [comment]
@timuralp Please review.

File: src/main/java/org/gaul/s3proxy/ChunkedInputStream.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            if (line.equals("")) {
+                return -1;
+            }
+            String[] parts = line.split(";", 2);
+            currentLength = Integer.parseInt(parts[0]);
+            currentSignature = parts[1];
+            chunk = new byte[currentLength];
+            currentIndex = 0;
+            ByteStreams.readFully(in, chunk);
+            // TODO: check chunkSinature
```
Comment (timuralp) [inline src/main/java/org/gaul/s3proxy/ChunkedInputStream.java]: chunkSignature

Comment (andrewgaul) [inline src/main/java/org/gaul/s3proxy/ChunkedInputStream.java]: Done.

**Comment by timuralp** [comment]
This looks fine to me. Too bad the s3-tests project doesn't have v4 signer tests, but hopefully they'll move faster on that than they have on other issues. The SDK tests should help in the meantime.

**Comment by andrewgaul** [comment]
s3-tests does have tests although I cannot run them as discussed in https://github.com/ceph/s3-tests/pull/84#issuecomment-170142987.

**Pull request merged**